### ISSUE TITLE
Scripts in Examples/Python and notebooks should check the version of the pip installed modules

### ISSUE THREAD

**Issue by akenmorris** [SCIInstitute/ShapeWorks#1135]
I don't think we currently have any versioning on the pip installed modules.  If a user upgrading ShapeWorks forgets to re-run conda_installs, there's really no check for this, things will just break in various ways with unhelpful error messages.  Seems like we could just add the ShapeWorks version (e.g. 6) to the modules and check it in RunUseCase.py.  Even better would be an incrementing number so that this doesn't happen with interim development releases for users as well.

**Comment by cchriste** [comment]
This is an important item, but I don't want our examples and notebooks to have to check for version numbers. It wouldn't be done for any other imports. Would improving conda_installs to check before it re-installs the world be an option to achieve the same result?

**Comment by cchriste** [comment]
Also, is this something we can push back to 6.2?

**Comment by cchriste** [comment]
semi-related to #1282

**Comment by akenmorris** [comment]
Moving this to deployment & infrastructure as we don’t need it for a particular version and don’t have a plan yet.

**Comment by cchriste** [comment]
One important component of this is to ensure versions are specified for even package installed by `shapeworks_install.sh` (is .bat needed anymore? I noticed the GitHub issue uses .sh, and from windows I just used .sh; it's also out-of-date; still necessary for users' installs, right? If so, GitHub action should probably use it).

But in general, I think this particular issue should be closed or at the most renamed to "add versions to pip shapeworks pip packages" since we developers should be sources `devenv.sh`(or .bat) and these will always use the current version in their source trees. I'm just not sure why we would have notebooks/scripts out-of-sync with the installation or the current source tree.

**Comment by akenmorris** [comment]
>  I'm just not sure why we would have notebooks/scripts out-of-sync with the installation or the current source tree.

The potential issue arises when a user installs a new version of ShapeWorks and forgets to run `install_shapeworks.sh/bat`.  When they run the included RunUseCase.py scripts, they will almost certainly get a fairly unintelligible error that gives them no indication of what went wrong (forgot to run install script).  It would just be nice if the script could check the version something and instruct the user.

**Comment by cchriste** [comment]
Fair enough. Let's add versions to our pip modules (and the shapeworks module itself, and the command line), and incorporate checks in the use cases. 

I'd still avoid too much boilerplate like this outside the use cases (e.g., in the notebooks) since we want users to learn from simple examples, and version checks everywhere would result in a big ol' cut n' paste party for everyone. 

With that preamble, whoever draws the lucky straw on this issue, please _just add one function to verify versions_, preferably called outside the tests themselves (i.e., call it from RunUseCase).

**Comment by cchriste** [comment]
```
(sw) $ shapeworks --version
ShapeWorks Version 6.3.0-dev
```
This uses Configuration.h, created by CMake file in the Applications folder using values specified in the global CMakeLists.txt. This string (or its composite digits, also specified therein) would be useful. Though as mentioned earlier, relying on only these during development might cause just as many invisible version mismatches. Another issue is that the `setup.py` files in all of the modules specify their own versions, none of which correlate with the ShapeWorks version.

In summary:
- no version in the Python module, which would be desirable
- independent, unrelated versions in all of the modules' setup.py 
- versions are insufficiently granular to assist development
